### PR TITLE
Add vars and tasks to support VGAuth guest operation testing

### PR DESCRIPTION
**Pull request by keirazhang**
This fix includes:
1. added common tasks to gather or manage vCenter Server domain user
2. enabled VGAuthService debug logging on Linux and Windows guest OS
3. collected VGAuthService log file on Linux and Windows guest OS

### CODE REVIEW

File: common/vcenter_get_domain_user_info.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    validate_certs: "{{ validate_certs | default(false) }}"
+    domain: "{{ vcenter_domain_name }}"
+    search_string: "{{ vcenter_domain_name }}\\{{ vcenter_domain_user_name }}"
+    exact_match: true
+  register: get_domain_user_result
+
+- name: "Set fact of vCenter domain user existing or not"
```
Comment (123lzxm) [inline common/vcenter_get_domain_user_info.yml:42]: "Set fact of vCenter domain user existing or not" --> "Set fact of vCenter domain user"?

File: common/vcenter_manage_domain_user.yml [hunk 2]
```diff
@@ -0,0 +1,121 @@
+# Copyright 2023 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Add or delete a domain user on vCenter Server
```
Comment (123lzxm) [inline common/vcenter_manage_domain_user.yml:4]: Maybe indicate which input parameter is required here. 
Or move line 27 task  to the top. Then we can clearly konw which input parameter is required.

File: common/vcenter_manage_domain_user.yml [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  ansible.builtin.set_fact:
+    manage_domain_user_cmd: >-
+      /usr/lib/vmware-vmafd/bin/dir-cli user delete
+      --account "{{ vcenter_domain_user_name }}"
+      --login "{{ vcenter_username }}"
+      --password "{{ vcenter_password }}"
+  when: vcenter_domain_user_op == "delete"
+
+- name: "{{ vcenter_domain_user_op | capitalize }} domain user '{{ vcenter_domain_user_name }}@{{ vcenter_domain_name }}'"
+  ansible.builtin.command: "{{  manage_domain_user_cmd }}"
```
Comment (123lzxm) [inline common/vcenter_manage_domain_user.yml:67]: {{  manage_domain_u, it seems there are two Blanks here.

File: linux/utils/enable_vmtools_logging.yml [hunk 4]
```diff
@@ -72,6 +72,13 @@
   ansible.builtin.debug: var=vmtools_config
   when: enable_debug
 
+- name: "Collect VMware Tools config file to local log directory"
+  include_tasks: fetch_file.yml
+  vars:
+    fetch_file_src_path: "{{ vmtools_config_file }}"
+    fetch_file_dst_path: "{{ current_test_log_folder }}/"
```
Comment (123lzxm) [inline linux/utils/enable_vmtools_logging.yml:79]: no need "/"

File: windows/utils/win_enable_vgauth_log.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    line: "loglevel=verbose"
+    state: present
+  delegate_to: "{{ vm_guest_ip }}"
+  when: win_check_file_exist_result
+
+- name: "Save VGAuthService config file to local log directory"
+  include_tasks: win_get_file.yml
+  vars:
+    win_get_file_src_path: "{{ vgauth_config_file }}"
+    win_get_file_dst_path: "{{ current_test_log_folder }}/"
```
Comment (123lzxm) [inline windows/utils/win_enable_vgauth_log.yml:47]: no need "/"

File: windows/utils/win_collect_vgauth_logs.yml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+      include_tasks: win_execute_cmd.yml
+      vars:
+        win_powershell_cmd: "Stop-Service -Name VGAuthService"
+      when: vgauth_service_status == "Running"
+
+    - name: "Get VGAuthServce log file"
+      include_tasks: win_get_file.yml
+      vars:
+        win_get_file_src_path: "{{ vgauth_log_file_src }}"
+        win_get_file_dst_path: "{{ current_test_log_folder }}/"
```
[the remark this answers is not in the document]
Comment (keirazhang) [reply windows/utils/win_collect_vgauth_logs.yml]: Removed.

File: windows/utils/win_enable_vgauth_log.yml [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+      [localization]
+      msgCatalog=C:\\ProgramData\\VMware\\VMware VGAuth\\msgCatalogs\\
+  when: not win_check_file_exist_result
+
+- name: "Update VGAuthService config file with logging options"
+  community.windows.win_lineinfile:
+    path: "{{ vgauth_config_file }}"
+    regexp: "^(# *)?loglevel.*"
+    line: "loglevel=verbose"
+    state: present
```
[the remark this answers is not in the document]
Comment (keirazhang) [reply windows/utils/win_enable_vgauth_log.yml]: Removed.

File: windows/utils/win_enable_vgauth_log.yml [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+- name: "Set facts of VGAuthService config file"
+  ansible.builtin.set_fact:
+    vgauth_config_file: "C:\\ProgramData\\VMware\\VMware VGAuth\\vgauth.conf"
+
+- name: "Check VGAuthService config file exists or not"
+  include_tasks: win_check_file_exist.yml
+  vars:
+    win_check_file_exist_file: "{{ vgauth_config_file }}"
+
+- include_tasks: win_write_to_file.yml
```
[the remark this answers is not in the document]
Comment (keirazhang) [reply windows/utils/win_enable_vgauth_log.yml]: Fixed. Thanks

File: env_setup/env_setup.yml [hunk 9]
```diff
@@ -67,17 +67,8 @@
...
-        add_host_in_memory_inventory_ip: "{{ esxi_hostname }}"
-        add_host_in_memory_inventory_gp: "esxi"
-        add_host_in_memory_inventory_user: "{{ esxi_username }}"
-        add_host_in_memory_inventory_pwd: "{{ esxi_password }}"
-        add_host_in_memory_inventory_python: "/bin/python"
-        add_host_in_memory_inventory_ssh_pipeline: true
-        add_host_in_memory_inventory_remote_tmp: "{{ vm_datastore_path }}"
-        add_host_in_memory_inventory_shell: "{{ esxi_shell }}"
+    - name: "Add vSphere hosts into memory inventory"
+      include_tasks: ../common/add_vsphere_hosts_in_inventory.yml
```
[the remark this answers is not in the document]
Comment (keirazhang) [reply env_setup/env_setup.yml:71]: We use vsphere_hostname to represent vCenter Server or ESXi server hostname. I don't think it is not clear enough to use vsphere_hosts to represent vCenter Server or/and ESXi server hosts here.

File: linux/utils/enable_vgauth_logging.yml [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+    option_value: "{{ item.value }}"
+    ini_state: present
+  loop: "{{ vgauth_logging_options | replace('=',': ') | from_yaml | dict2items }}"
+  when: guest_file_exists
+
+- name: "Save VGAuthService config file to local log directory"
+  include_tasks: fetch_file.yml
+  vars:
+    fetch_file_src_path: "{{ vgauth_config_file }}"
+    fetch_file_dst_path: "{{ current_test_log_folder }}/"
```
[the remark this answers is not in the document]
Comment (keirazhang) [reply linux/utils/enable_vgauth_logging.yml:71]: Linux is different from windows while using ansible.builtin.fetch module. To add '/' explicitly is to fetch the file and put it directly under test case log folder, no need to recursively fetch its parent directories. See https://docs.ansible.com/ansible/latest/collections/ansible/builtin/fetch_module.html#parameter-flat.

File: common/vcenter_get_domain_user_info.yml [hunk 11]
```diff
@@ -0,0 +1,50 @@
+# Copyright 2023 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Gather information about a domain user on vCenter Server
+# Parameters:
+#   vcenter_domain_name: The vCenter Server user domain name. Default is same with vcenter_username domain.
+#   vcenter_domain_user_name: The domain user name.
+#
+- name: "Assert vcenter_domain_user_name is set"
```
[the remark this answers is not in the document]
Comment (keirazhang) [reply common/vcenter_get_domain_user_info.yml]: Done.

File: common/vcenter_get_domain_user_info.yml (hunk 1 above)
Comment (keirazhang) [reply common/vcenter_get_domain_user_info.yml]: Fixed.

File: common/vcenter_manage_domain_user.yml (hunk 2 above)
Comment (keirazhang) [reply common/vcenter_manage_domain_user.yml:4]: Added (optional) to optional parameters

File: common/vcenter_manage_domain_user.yml (hunk 3 above)
Comment (keirazhang) [reply common/vcenter_manage_domain_user.yml]: Fixed.

File: linux/utils/enable_vmtools_logging.yml (hunk 4 above)
Comment (keirazhang) [reply linux/utils/enable_vmtools_logging.yml:79]: Using '/' is to set 'flat=true' for linux fetching file.

File: linux/utils/enable_vgauth_logging.yml [hunk 12]
```diff
@@ -0,0 +1,101 @@
+# Copyright 2023 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Enable debug logging for VGAuthService within Linux guest OS
+# See https://kb.vmware.com/s/article/1007873 for details.
+#
+- name: "Set facts about VGAuthService"
+  include_tasks: set_vgauth_facts.yml
+
+- name: "Set default VGAuthService debug log directory"
```
[the remark this answers is not in the document]
Comment (keirazhang) [reply linux/utils/enable_vgauth_logging.yml:10]: I want to collect clean vgauth logs when testing vgauth guest operations

**Review by 123lzxm — APPROVED** [review]
LGTM, thanks.

**Pull request merged**